include/nbl/builtin/hlsl/bxdf/reflection/beckmann.hlsl

```diff
@@ -175,6 +184,109 @@ struct SBeckmannBxDF
         else
             return hlsl::promote<spectral_type>(0.0);
     }
+
+    sample_type generate(NBL_CONST_REF_ARG(isotropic_interaction_type) interaction, NBL_CONST_REF_ARG(vector2_type) u, NBL_REF_ARG(isocache_type) cache)
+    {
+        SBeckmannAnisotropicBxDF<Config> beckmann_aniso = SBeckmannAnisotropicBxDF<Config>::create(A, A, ior0, ior1);
+        anisocache_type anisocache;
+        sample_type s = beckmann_aniso.generate(anisotropic_interaction_type::create(interaction), u, anisocache);
+        cache = anisocache.iso_cache;
+        return s;
+    }
+
+    scalar_type pdf(NBL_CONST_REF_ARG(params_isotropic_t) params, NBL_REF_ARG(scalar_type) onePlusLambda_V)
+    {
+        SBeckmannDG1Query<scalar_type> dg1_query;
+        dg1_query.maxNdotV = params.getNdotV();
+    
+        scalar_type a2 = A*A;
+        ndf::Beckmann<scalar_type, false> beckmann_ndf;
+        beckmann_ndf.a2 = a2;
```

Comment:
why are you storing `A` as a member when you could store the parametrized NDF instead ?

btw seems that CookTorrance is a good idea as there's really not much difference between GGX and beckmann BRDF and BTDF

Reply:
continued in https://github.com/Devsh-Graphics-Programming/Nabla/pull/916/files#r2309532126